### DISCUSSION TITLE
V5: dynamic context types

### DISCUSSION THREAD

**Discussion by gkiely** [statelyai/xstate#4473] [Q&A]
I'm following this step on machine with context https://stately.ai/docs/migration#set-context-with-input-not-machinewithcontext

Is it possible to set the types for a dynamic context, I would like it to return type `Data` instead of `unknown` in the below example. `Data` could be many different types using this machine so I do want to hard code it in the machine.

Sandbox: https://codesandbox.io/s/rough-leftpad-2gscs2?file=/src/App.tsx:85-285

**Comment by Andarist** [comment]
You need to make `input` (or just `data`) generic. The type that you are reaching is embedded in `throttleFetchMachine` and we can't retroactively put `Data` in it where you are using `useMachine`. TS requires top-down flow of the types.

The most straightforward way of doing this looks like this: https://codesandbox.io/s/thirsty-dhawan-d6z8py?file=/src/App.tsx
